prog/encodingexec.go

```diff
@@ -68,6 +112,44 @@ func (p *Prog) SerializeForExec(buffer []byte, pid int) error {
 				})
 			}
 		})
+		// Generate checksum calculation instructions starting from the last one,
+		// since checksum values can depend on values of the latter ones
+		var csumArgs []*Arg
+		for arg, _ := range csumMap {
+			csumArgs = append(csumArgs, arg)
+		}
+		sort.Sort(ByPhysicalAddr{Args: csumArgs, Context: w})
```

Comment:
Put this all under `if len(csumMap) > 0`, this interface conversion will also allocate.